Feature: Custom-Color-Themer

willwade opened the pull request:
Creates a big modal window. Not sure about this. @ms-mialingvo take a look 

attempt at a #583

willwade committed:
attempt at a #583

Creates a big modal window. Not sure about this

willwade commented:
@ms-mialingvo  https://gridbeta-github.asterics-foundation.org/willwade/598-merge

klues reviewed (changes requested):
Thanks for your work! 👍
I've reviewed the code and added quite some comments. I hope they're understandable - especially the very long one about data flow 😉

willwade committed:
Fix button sizes and layout for custom color schemes

willwade committed:
Use standard button styling for custom color scheme buttons

willwade committed:
fix button sizing

willwade committed:
add cancel

willwade committed:
Remove dependency on a non-existent helper in the editor & Ensure active scheme stays valid after deletes

Ensure active scheme stays valid after deletes - If a custom scheme gets deleted in the modal and it was the active one, we now fall back to the default scheme before saving.
Replaced constants.createCustomColorScheme(...) with a small local builder that creates categories for the selected base and an all-white palette (and adds Fitzgerald mappings when relevant). This makes the editor robust even if it initializes in “create” mode.

MinecraftMoments commented:
this is very good visually! i think being able to create custom categories would also be nice (since i don't use purple for places, places are nouns, and i use the "place" category / purple color for questions instead), but other than that, this already works basically exactly as i requested!!

willwade commented:
> this is very good visually! i think being able to create custom categories would also be nice (since i don't use purple for places, places are nouns, and i use the "place" category / purple color for questions instead), but other than that, this already works basically exactly as i requested!!

Yeah I had done that but its really complicated - we settled so far on duplicating an existing theme and editing that than doing it by scratch due to the various different ways you'd have to categorise  button types. Maybe we release it as is and think how we do this for a future Issue

arasaac-dga commented:
Works as expected! Why is "Base Scheme Type" deactivated in the modal window? It’s confusing to see it disabled while knowing that the modal window actually uses the active color scheme. We think it would be better if it were active, so you could change the color scheme directly in the modal window from the one you want to start with.

willwade committed:
shows the base theme so you can change this in the editing modal

willwade committed:
allow save button to work :)

@arasaac-dga are fast on their reviews!

willwade committed:
fix duplicate creation on save

Fukurouonthesea commented:
> > this is very good visually! i think being able to create custom categories would also be nice (since i don't use purple for places, places are nouns, and i use the "place" category / purple color for questions instead), but other than that, this already works basically exactly as i requested!!
> 
> Yeah I had done that but its really complicated - we settled so far on duplicating an existing theme and editing that than doing it by scratch due to the various different ways you'd have to categorise button types. Maybe we release it as is and think how we do this for a future Issue

(just reading through things: could you rename existing categories as a mid step? instead of creating new ones. Maybe that'd be easier.
Example - I use (place) purple for all socials, and (social) pink for medical, and so on. It works, but would be nicer to be able to assign them without remembering the "translations".)
